invenio_app_ils/records/permissions.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2018 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Ils records' permissions."""
+from flask_principal import ActionNeed, RoleNeed, UserNeed
+from invenio_access import Permission, superuser_access
+
+librarian_role = RoleNeed('librarian')
+create_records_action = ActionNeed('create-records')
+
+
+class RecordPermission(Permission):
+    """Record permission.
+
+    - Create action given to librarian, admin and specified users.
+    - Read access given to everyone with possibility to hide.
+    - Delete access to admin and specified users.
+    """
+
+    create_actions = ['create']
+    read_actions = ['read']
+    update_actions = ['update']
+    delete_actions = ['delete']
+
+    def __init__(self, record, action):
+        """Constructor."""
+        self.record = record
+        self.action = action
+        self._permissions = None
+        record_needs = self._dispatch()
+        super(RecordPermission, self).__init__(*record_needs)
+
+        # removes need of superuser for reading access
+        if self.allow_by_default:
```

Comment:
I don't get this! 😨

Reply:
just for leaving a note because we have discuss about it: the line removes the superuser-access from the set of Needs